### PR TITLE
Prepare `v0.12.4`

### PR DESCRIPTION
**Pull request by AurevoirXavier**

### CODE REVIEW

**Review by HackFisher — APPROVED** [review]

**Comment by AsceticBear** [comment]
The comparison `darwinia-common` : https://github.com/darwinia-network/darwinia-common/compare/darwinia-v0.12.3...darwinia-network:darwinia-v0.12.4

**Comment by AsceticBear** [comment]
After thinking it over again, I need we need to apply the new KTON precompile in this release. Because the original `Transfer` precompile has been disabled, that means the old kton transfer way is unavailable. @hujw77 @AurevoirXavier @xiaoch05 @1022ranji  @ZolaxZola。We need to stabilize the new `KTON` solution before the next upgrade and make sure that all relevant front and back-end applications are updated.

File: runtime/darwinia/src/lib.rs [hunk 1]
```diff
@@ -261,7 +261,7 @@ frame_support::construct_runtime! {
 		EthereumRelay: darwinia_bridge_ethereum::{Pallet, Call, Storage, Config<T>, Event<T>} = 32,
 		EthereumBacking: to_ethereum_backing::{Pallet, Call, Storage, Config<T>, Event<T>} = 31,
 		EthereumRelayerGame: darwinia_relayer_game::<Instance1>::{Pallet, Storage} = 33,
-		EthereumRelayAuthorities: darwinia_relay_authorities::<Instance1>::{Pallet, Call, Storage, Event<T>} = 36,
+		EthereumRelayAuthorities: darwinia_relay_authority::{Pallet, Call, Storage, Event<T>} = 36,
```
Comment (AsceticBear) [inline runtime/darwinia/src/lib.rs:264]: Do we need migration here? Check out https://github.com/darwinia-network/darwinia-common/pull/1335/files#diff-acda1afbc988e35deb9fd2d5892ebe7751542c8de3eeb617fd13011154a67ac0

File: runtime/darwinia/src/pallets/relayer_game.rs [hunk 2]
```diff
@@ -4,8 +4,8 @@ pub use darwinia_relayer_game::Instance1 as EthereumRelayerGameInstance;
 use frame_support::traits::LockIdentifier;
 // --- darwinia-network ---
 use crate::*;
-use darwinia_relay_primitives::AdjustableRelayerGame;
```
Comment (AsceticBear) [inline runtime/darwinia/src/pallets/relayer_game.rs:7]: Another place related to this companion is:

```rs
Call::EthereumBacking(..) | Call::EthereumRelay(..) | Call::EthereumRelayAuthorities(..)
```

Or

```rs
Call::EthereumBacking(..) | Call::EthereumRelay(..) | Call::EcdsaRelayAuthority(..)
```

Please check it again.

File: runtime/crab/src/migrations.rs [hunk 3]
```diff
@@ -23,6 +23,28 @@ impl OnRuntimeUpgrade for CustomOnRuntimeUpgrade {
 }
 
 fn migrate() -> Weight {
-	0
-	// RuntimeBlockWeights::get().max_block
+	use frame_support::StorageHasher;
+
+	// TODO
```
Comment (AsceticBear) [inline runtime/crab/src/migrations.rs:28]: Waiting for the official deployed Kton migrated contract.

File: runtime/darwinia/src/pallets/to_ethereum_backing_.rs [hunk 4]
```diff
@@ -16,7 +16,7 @@ frame_support::parameter_types! {
 
 impl Config for Runtime {
 	type AdvancedFee = AdvancedFee;
-	type EcdsaAuthorities = EthereumRelayAuthorities;
+	type EcdsaRelayAuthority = EthereumRelayAuthorities;
```
Comment (AsceticBear) [inline runtime/darwinia/src/pallets/to_ethereum_backing_.rs:19]: This is different from the Pangolin/Pangoro config. Please check it again.

**Comment by AurevoirXavier** [comment]
Rename `EthereumRelayAuthorities` to `EcdsaRelayAuthority`. cc @fewensa

**Comment by AurevoirXavier** [comment]
Ready to merge? @AsceticBear

**Review by AsceticBear — APPROVED** [review]
LGTM

**Pull request merged**